File: IFC/1.3/programas/perfeito.c
/*
* Autor: Carlos Antunis Bonfim da Silva Santos
* Sobre:
* Programa para testar se um número é perfeito.
*/
#include<stdio.h>
int main()
{
    unsigned numero, soma_dos_fatores=0;

    // Obter o número
    printf("Informe o número: ");
    scanf("%u", &numero);

    for (unsigned fator = 1; fator < numero; fator++) {
        if (numero % fator == 0) {  /*Se um fator inteiro divide o número
                                    * então some na soma dos divisores*/
        printf("%u\n", fator);
            soma_dos_fatores += fator;
        }
    }

    if (soma_dos_fatores == numero) {   /*Se a soma dos divisores for exatamente
                                        * o número, ele é um número perfeito*/
        printf("%u é perfeito\n", numero);
    } else { //Caso contrário ele não é perfeito
        printf("%u não é perfeito\n", numero);
    }

    return 0;
}

File: 1.IFC/Codes/Python/Perfeito.py
#Autor: Carlos Antunis Bonfim da Silva Santos
#Objetivo:
#Computar se um número é perfeito

#Obter o número
Numero = int(input("Informe o número a ser testado: "))

#Somar todos os fatores do Número
Soma_dos_Fatores = 0;
for fator in range(1,Numero):
    if (Numero % fator == 0):
        Soma_dos_Fatores = Soma_dos_Fatores + fator

if (Numero == Soma_dos_Fatores): #Se o número for
                                 #a soma dos fatores
                                 #ele é perfeito
    print(Numero, "é perfeito")
else:
    print(Numero, "não é perfeito")